lightningd/build_utxos.c

```diff
@@ -179,7 +179,12 @@ static void unreserve_utxo(struct lightningd *ld, const struct utxo *unres)
 {
 	struct tracked_utxo *utxo;
 
-	list_for_each(&ld->utxos, utxo, list) {
+	assert(wallet_update_output_status(ld->wallet, &unres->txid,
+					   unres->outnum, output_state_reserved,
+					   output_state_available));
```

Comment:
Outside testing code, it's bad form to put code with side-effects inside assert().  That's because with NDEBUG defined, the assert.h header does `#define assert(x)` which removes this entirely.

You can fatal() here, instead.